src/log/accumulators.jl

```diff
@@ -0,0 +1,208 @@
+
+
+
+@inline _prod(x::T, y::T) where T = x * y
+@inline _prod(x::Complex, y::Complex) = Complex(real(x) * real(y), imag(x) * imag(y))
+@inline _prod(x::AbstractArray, y::AbstractArray) = _prod.(x, y)
+
+
+
+"""
+    FastVariance <: AbstractVarianceAccumulator
+
+The `FastVariance` accumulator keeps track of `∑x` and `∑x²` where `x` refers to
+the pushed values. It is significantly faster than the `Variance` accumulator,
+but can become unstable for some inputs. (For example when the mean is much 
+large than the variance of the pushed values.)
+
+See also: [`AbstractVarianceAccumulator`](@ref), [`Variance`](@ref)
+"""
+mutable struct FastVariance{T} <: AbstractVarianceAccumulator{T}
+    x_sum::T
+    x2_sum::T
+    count::Int
+end
+function FastVariance{T}(T0) where T 
+    FastVariance{T}(deepcopy(T0), deepcopy(T0), zero(Int))
+end
+FastVariance{T}() where T = FastVariance(zero(T), zero(T), zero(Int))
+FastVariance() = FastVariance{Float64}()
+
+@inline Base.isempty(V::FastVariance) = iszero(V.count)
+function Base.empty!(V::FastVariance)
+    V.x_sum, V.x2_sum = zero(V.x_sum), zero(V.x2_sum)
+    V.count = zero(Int)
+    V
+end
+Base.:(==)(a::FastVariance{T}, b::FastVariance{T}) where T = (
+    (a.x_sum == b.x_sum)
+    && (a.x2_sum == b.x2_sum)
+    && (a.count == b.count)
+)
+function Base.copy(V::FastVariance{T}) where T
+    FastVariance{T}(copy(V.x_sum), copy(V.x2_sum), copy(V.count))
+end
+
+mean(V::FastVariance) = isempty(V) ? zero(V.x_sum) : V.x_sum / V.count
+
+function var(V::FastVariance{T}) where {T <: Real}
+    n, X, X2 = V.count, V.x_sum, V.x2_sum
+    X2 / (n - 1) - X^2 / (n*(n - 1))
+end
+
+function var(V::FastVariance{T}) where {T <: Complex}
+    n, X, X2 = V.count, V.x_sum, V.x2_sum
+    (real(X2) + imag(X2)) / (n - 1) - abs2(X) / (n*(n - 1))
+end
+
+function var(V::FastVariance{<: AbstractArray{T, D}}) where {D, T <: Real}
+    n, X, X2 = V.count, V.x_sum, V.x2_sum
+    @. X2 / (n - 1) - X^2 / (n*(n - 1))
+end
+
+function var(V::FastVariance{<: AbstractArray{T, D}}) where {D, T <: Complex}
+    n, X, X2 = V.count, V.x_sum, V.x2_sum
+    @. (real(X2) + imag(X2)) / (n - 1) - abs2(X) / (n*(n - 1))
+end
+
+
+function Base.push!(V::FastVariance{T}, value::T) where T
+    V.x_sum += value
+    V.x2_sum += _prod(value, value)
+    V.count += 1
+    return V
+end
+
+function Base.push!(V::FastVariance{T}, value::T) where T <: AbstractArray
+    V.x_sum .+= value
+    @. V.x2_sum += _prod(value, value)
+    V.count += 1
+    return V
+end
+
+
+
+"""
+    Variance <: AbstractVarianceAccumulator
+
+The `Variance` accumulator uses Welfords algorithm to keep track of pushed 
+values and to compute variances and errors. It is less error prone than 
+`FastVariance` but also slower.
+
+See also: [`AbstractVarianceAccumulator`](@ref), [`FastVariance`](@ref)
+"""
+mutable struct Variance{T} <: AbstractVarianceAccumulator{T}
+    δ::T
+    m1::T
+    m2::T
+    count::Int
+end
```

Comment:
Bit awkward to have `δ` in there and only use it for Arrays but it does shave of the remaining allocations. Array binning is quite a lot faster, number binning seems to be unaffected.